superset/assets/src/visualizations/Histogram.jsx

```diff
@@ -0,0 +1,131 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Histogram, BarSeries, XAxis, YAxis } from '@data-ui/histogram';
+import { chartTheme } from '@data-ui/theme';
+import { LegendOrdinal } from '@vx/legend';
+import { scaleOrdinal } from '@vx/scale';
+import { getColorFromScheme } from '../modules/colors';
+import './histogram.css';
+
+const propTypes = {
+  className: PropTypes.string,
+  data: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string,
+    values: PropTypes.arrayOf(PropTypes.number),
+  })).isRequired,
+  width: PropTypes.number.isRequired,
+  height: PropTypes.number.isRequired,
+  colorScheme: PropTypes.string,
+  normalized: PropTypes.bool,
+  binCount: PropTypes.number,
+  opacity: PropTypes.number,
+  xAxisLabel: PropTypes.string,
+  yAxisLabel: PropTypes.string,
+};
+const defaultProps = {
+  className: '',
+  colorScheme: '',
+  normalized: false,
+  binCount: 15,
+  opacity: 1,
+  xAxisLabel: '',
+  yAxisLabel: '',
+};
+
+class CustomHistogram extends React.PureComponent {
+  render() {
+    const {
+      className,
+      data,
+      width,
+      height,
+      binCount,
+      colorScheme,
+      normalized,
+      opacity,
+      xAxisLabel,
+      yAxisLabel,
+    } = this.props;
+
+    const keys = data.map(d => d.key);
+    const colorScale = scaleOrdinal({
+      domain: keys,
+      range: keys.map(key => getColorFromScheme(key, colorScheme)),
+    });
+
+    return (
+      <div className={`histogram-chart ${className}`}>
+        <div className="legend-container">
+          <LegendOrdinal
+            scale={colorScale}
+            direction="row"
+            shape="rect"
```

Comment:
I could see `rect` being better than circles given that bars are rectangle. If you _do_ want them to render as rects, you can do something like (example [here](https://github.com/williaster/data-ui/blob/master/packages/demo/examples/01-xy-chart/AreaDifferenceSeriesExample.jsx) / [storybook](https://williaster.github.io/data-ui/?selectedKind=xy-chart&selectedStory=AreaDifferenceSeries&full=0&addons=0&stories=1&panelRight=0)). 

tldr

```javascript
shape={({ fill, width, height }) => (
  <svg width={width} height={height}>
    <rect width={width} height={height} fill={fill} />
  </svg>
)}
```

Reply:
Hah. nice.